faraday/deviceconfiguration.py

```diff
@@ -0,0 +1,440 @@
+        path = location
+        break
+    except ConfigParser.NoSectionError:
+        pass
+
+logger = logging.getLogger('DeviceConfiguration')
+
+# Create Device Configuration configuration file path
+deviceConfigurationConfigPath = os.path.join(path, "deviceconfiguration.ini")
+faradayConfigPath = os.path.join(path, "faraday_config.ini")
+logger.debug('deviceconfiguration.ini PATH: ' + deviceConfigurationConfigPath)
+logger.debug('faraday_config.ini PATH: ' + faradayConfigPath)
+
+# Load Device Configuration Configuration from deviceconfiguration.ini file
+deviceConfig = ConfigParser.RawConfigParser()
+
+# Command line input
+parser = argparse.ArgumentParser(description='Device Configuration application provides a Flask server to program Faraday radios via an API')
+parser.add_argument('--init-config', dest='init', action='store_true', help='Initialize Device Configuration configuration file')
+parser.add_argument('--init-faraday-config', dest='initfaraday', action='store_true', help='Initialize Faraday configuration file')
+parser.add_argument('--callsign', help='Set Proxy Faraday callsign to connect to and program')
+parser.add_argument('--nodeid', type=int, help='Set Proxy Faraday nodeid to connect to and program')
+parser.add_argument('--faradayconfig', action='store_true', help='Display Faraday configuration file contents')
+
+# Faraday Configuration
+parser.add_argument('--fcallsign', help='Set Faraday radio callsign')
+parser.add_argument('--fnodeid', type=int, help='Set Faraday radio nodeid')
+parser.add_argument('--fconfigboot', action='store_false', help='Set Faraday radio config boot bit OFF')
+parser.add_argument('--fgpiop3', type=int, help='Set Faraday radio fgpio_p3')
+parser.add_argument('--fgpiop4', type=int, help='Set Faraday radio fgpio_p4')
+parser.add_argument('--fgpiop5', type=int, help='Set Faraday radio fgpio_p5')
+parser.add_argument('--fbootfrequency', type=float, help='Set Faraday radio boot frequency')
+parser.add_argument('--fbootrfpower', type=int, help='Set Faraday radio boot RF power')
+parser.add_argument('--flatitude', type=float, help='Set Faraday radio default latitude. Format \"ddmm.mmmm\"')
+parser.add_argument('--flongitude', type=float, help='Set Faraday radio default longitude. Format \"dddmm.mmmm\"')
+parser.add_argument('--flatitudedir', help='Set Faraday radio default latitude direction (N/S)')
+parser.add_argument('--flongitudedir', help='Set Faraday radio default longitude direction (E/W)')
+parser.add_argument('--faltitude', type=float, help='Set Faraday radio default altitude in meters. Maximum of 17999.99 Meters')
+# Purposely do not allow editing of GPS altitude units
+parser.add_argument('--fgpsboot', action='store_false', help='Set Faraday radio GPS boot power OFF')
+parser.add_argument('--fgps', action='store_true', help='Set Faraday radio GPS use ON')
+parser.add_argument('--fuarttelemetry', action='store_false', help='Set Faraday radio UART Telemetry OFF')
```

Comment:
How do you toggle back on? I suggest either allowing argument options 0 and '1' or creating an ON and OFF function arguments.